### PR TITLE
Add atmosphere DAG analysis to the AD

### PR DESCRIPTION
**Pull request by bartgol**
This PR does some minor cleaning, and then mostly focuses on adding a capability to the AD. Namely, the AD can now build a Directed Acyclic Graph (DAG) of the atmosphere, and check that all the dependencies are met.

In the atm_process_tests, I added a test that creates a dag of a dummy atm configuration, and writes a .dot file, that graphviz can then process into a jpg file (with the `dot` command). The current strategy is to add grids remappers to the graph (with a name linked to the atm proc that required/computed the remapped fields). Coreography changes suggestions are welcome.

Note: there could be more to add to this PR, but I'd rather merge a small PR now and another in a few weeks, than leaving this work outstanding for a while.

### CODE REVIEW

**Comment by bartgol** [comment]
Added WIP tag, cause the p3_shoc_coupled_ut test needs to be fixed (the AD identifies unmet field dependencies).

**Comment by bartgol** [comment]
Ok, I think this PR is ready.

@AaronDonahue I have other things I need/want to add, but it's been outstanding for a while, and I'd like to get some capabilities in now and put the rest later, rather than wait weeks until everything is done.

File: components/scream/src/physics/p3/scream_p3_interface.F90 [hunk 1]
```diff
@@ -57,26 +57,27 @@ subroutine p3_init_f90 () bind(c)
...
+    real(kind=c_real) :: T(pcols,pver)
+
     character(len=100) :: case_title
 
     integer(kind=c_int) :: i, k
     logical(kind=c_bool) :: masterproc
 
     ! READ inputs from SCM for p3-stand-alone:
-    q(:,:,:) = 0.0_rtype
+    qdp(:,:,:) = 0.0_rtype
```
Comment (AaronDonahue) [inline components/scream/src/physics/p3/scream_p3_interface.F90:80]: I believe the presence of `qdp` in the original p3_standalone code confused things.  The original intent was just to check that a variable from dynamics was making it into physics.  But P3 itself does not use qdp, it just uses q.  We can remove qdp and any reference to it entirely from the P3 code and revert all the `q` references back to just plain `q`.

Comment (bartgol) [reply components/scream/src/physics/p3/scream_p3_interface.F90:80]: Ok, I'm going to purge qdp completely.

Comment (bartgol) [reply components/scream/src/physics/p3/scream_p3_interface.F90]: Done!

**Pull request merged**